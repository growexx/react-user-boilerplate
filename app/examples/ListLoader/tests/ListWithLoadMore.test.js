import React from 'react';
import { fireEvent, render, wait } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import request from 'utils/request';
import configureStore from '../../../configureStore';
import ListWithLoadMore from '../ListWithLoadMore';
let store;
jest.mock('utils/request');
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <ListWithLoadMore />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );
describe('<ListWithLoadMore />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    request.mockImplementation(() => Promise.resolve({ status: 1 }));
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('call LoadMore', async () => {
    request.mockImplementation(() =>
      Promise.resolve({
        status: 1,
        results: [
          {
            name: {
              last: 'test',
              email: 'test@234.com',
            },
          },
        ],
      }),
    );
    const { container } = componentWrapper();
    await wait(() => expect(request).toHaveBeenCalledTimes(2));
    fireEvent.click(container.querySelector('button'));
    expect(request).toHaveBeenCalled();
  });
});
