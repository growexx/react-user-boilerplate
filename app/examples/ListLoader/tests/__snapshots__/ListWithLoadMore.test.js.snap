// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListWithLoadMore /> should render and match the snapshot 1`] = `
<div
  class="ant-list ant-list-split ant-list-loading demo-loadmore-list"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div>
      <div
        class="ant-spin ant-spin-spinning"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
    <div
      class="ant-spin-container ant-spin-blur"
    >
      <div
        style="min-height: 53px;"
      />
    </div>
  </div>
</div>
`;
